src/Arcus.Scripting.SQL/Scripts/RunDatabaseScript.ps1

```diff
@@ -0,0 +1,158 @@
+    [String]
+    $ServerName=$(throw "Please provide the name of the SQL Server that hosts the SQL Database. (Do not include 'database.windows.net'"),
+    
+    [Parameter(Mandatory=$True)]
+    [String]
+    $DatabaseName=$(throw "Please provide the name of the SQL Database"),
+    
+    [Parameter(Mandatory=$True)]
+    [String]
+    $UserName=$(throw "Please provide the UserName of the SQL Database"),
+    
+    [Parameter(Mandatory=$True)]
+    [String]
+    $Password=$(throw "Please provide the Password of the SQL Database")
+)
+
+Import-Module SqlServer
+
+Write-Host $PSScriptRoot
+$scriptpath = "$PSScriptRoot/sqlScripts"
+
+#Functions for repeated use
+function QueryDatabase($params, [String]$query)
+{
+    $result = Invoke-Sqlcmd @params -Query $query -Verbose -QueryTimeout 180
+    Write-Host "Result from querying $($result)"
+}
+
+function QueryDatabaseWithResult($params, [String]$query)
+{
+    $result = Invoke-Sqlcmd @params -Query $query -Verbose
+    return $result
+}
+
+function CreateDbParams([String]$DatabaseName, [String]$serverInstance, [String]$UserName, [String]$Password)
+{
+    Write-Host "databasename = $DatabaseName"
+    Write-Host "serverinstance = $serverInstance"
+    Write-Host "username = $UserName"
+    
+    return $params = @{
+      'Database' = $DatabaseName
+      'ServerInstance' = "$serverInstance.database.windows.net"
+      'Username' = $UserName
+      'Password' = $Password
+      'OutputSqlErrors' = $true
+      'AbortOnError' = $True
+    }
+}
+
+function GetScriptFileText([String]$scriptPath, [String]$fileName)
+{
+    $currentfilepath = "$scriptPath/$fileName.sql"
+    return $query = [IO.File]::ReadAllText($currentfilepath)
+}
+
+#Group params needed to connect to database for ease of use
+$params = CreateDbParams $DatabaseName $ServerName $UserName $Password
+
+#Get all tables on the database
+$query = 'SELECT TABLE_NAME FROM information_schema.tables'
+$tables = QueryDatabaseWithResult $params $query
+
+# Get script to determine DB version
+$selectDbVersionScript = 'SELECT TOP 1 CurrentVersionNumber FROM DatabaseVersion ORDER BY CurrentVersionNumber DESC'
+
+#Create a DatabaseVersion table if it doesn't exist
+if(!$tables.ItemArray.Contains('DatabaseVersion'))
+{
+    Write-Host 'DatabaseVersion does not exist yet in this database.'
+    Write-Host 'Creating DatabaseVersion Table...'
+    $createDbQuery = GetScriptFileText $scriptpath 'CreateDatabaseVersionTable'
+    QueryDatabase $params $createDbQuery
+    Write-Host 'DatabaseVersion Table Created'
+
+    $databaseVersionNumber = 0
+}
+else
+{
+    # Get database versionnumber
+    Write-Host 'Getting Current Database Version Number...'
+    $databaseVersionNumberDataRow = QueryDatabaseWithResult $params $selectDbVersionScript
+
+    if ($databaseVersionNumberDataRow.ItemArray.Count -eq 0)
+	{
+		$databaseVersionNumber = 0;
+	}
+	else
+	{
+		$databaseVersionNumber = [convert]::ToInt32($databaseVersionNumberDataRow.ItemArray[0])
+	}
```

Comment:
As this is being re-used at the end of the script, perhaps it can be turned into a function?
Only 1 place to make a change, if required.

Reply:
Yes, added a function for this, can you check if it's ok? Thx.